### PR TITLE
Add User Guide

### PR DESCRIPTION
**Pull request by Rinder5**

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -41,138 +31,225 @@ By : `Team SE-EDU`  &nbsp;&nbsp;&nbsp;&nbsp; Since: `Jun 2016`  &nbsp;&nbsp;&nbs
...
+  you press on the 'up' arrow key. <br>
+  Shows 'add Test' in your input text field.
 
-* **Add**  `add NAME p/PHONE_NUMBER e/EMAIL a/ADDRESS [t/TAG]...` <br>
-  e.g. `add James Ho p/22224444 e/jamesho@gmail.com a/123, Clementi Rd, 1234665 t/friend t/colleague`
+### 2.12. Add alias for commands: 'add alias'
+
+Adds an alias for a command
+Format: 'addalias COMMAND ALIAS'
+
```
Comment (louislai) [inline docs/UserGuide.md]: `alias <alias> <command>` is more natural and is the more common option I believe. @Melvin-Tan might need discuss. There are a few ways we can tackle the `alias` command.
- One is to allow only aliasing of simple command, e.g `add`, `delete`, `find`
- Another option is to allow aliasing a whole phrase, e.g on unix u can do `alias v="mvim -v"
Supporting both is an option as well. The two scenarios are differentiated depending on there are `"` (quote) or no `"`. Personally I dont think it would be hugely difficult to do this

File: docs/UserGuide.md [hunk 2]
```diff
@@ -41,138 +31,225 @@ By : `Team SE-EDU`  &nbsp;&nbsp;&nbsp;&nbsp; Since: `Jun 2016`  &nbsp;&nbsp;&nbs
...
+### 2.13. Delete alias for commands: 'delete alias'
+
+Removes an alias for a command
+Format: 'deletealias ALIAS'
+
+### 2.14. View aliases for commands: 'view alias'
+
+Views aliases in the system
+Format: 'viewalias [COMMAND]'
```
Comment (louislai) [inline docs/UserGuide.md]: similarly i think `alias list` is more natural and common

File: docs/UserGuide.md [hunk 3]
```diff
@@ -1,35 +1,25 @@
...
+2. Type the command in the command box and press <kbd>Enter</kbd> to execute it. <br>
    e.g. typing **`help`** and pressing <kbd>Enter</kbd> will open the help window.
-5. Some example commands you can try:
-   * **`list`** : lists all contacts
-   * **`add`**` John Doe p/98765432 e/johnd@gmail.com a/John street, block 123, #01-01` :
-     adds a contact named `John Doe` to the Address Book.
-   * **`delete`**` 3` : deletes the 3rd contact shown in the current list
+3. Some example commands you can try:
+   * **`add`**` Try Out Todo List  d/11-12-2011` :
+     adds a task named `Try Out Todo List` to the todo list.
```
Comment (louislai) [inline docs/UserGuide.md]: Mismatched example. I think you mean that d/ is optional. 
In the future we can also put in `Natty`/`PrettyTime` to do date parsing.

I saw there was an `update INDEX [NAME] [d/ENDDATE] [sd/STARTDATE]` below. But seem like there isnt a corresponding add command for it

Personally I would prefer putting start date before end date also

File: docs/UserGuide.md [hunk 4]
```diff
@@ -183,4 +260,26 @@ There is no need to save manually.
 * **Select** : `select INDEX` <br>
   e.g.`select 2`
 
+* **Update** : '`update INDEX [NAME] [d/ENDDATE] [sd/STARTDATE] [r/PERIOD(DAY/WEEK/MONTH)] [t/TAG]...'
+  e.g. 'update 1 d/11/12/2011'
+
+* **Exit** : 'exit'
+
+* **Undo** : 'undo [NUMBER]'
+  e.g. 'undo 5'
```
Comment (louislai) [inline docs/UserGuide.md]: maybe good to provide a quick-hand command `undo` just to undo the last edit as well. Similarly for `redo`

File: docs/UserGuide.md [hunk 5]
```diff
@@ -183,4 +260,26 @@ There is no need to save manually.
 * **Select** : `select INDEX` <br>
   e.g.`select 2`
 
+* **Update** : '`update INDEX [NAME] [d/ENDDATE] [sd/STARTDATE] [r/PERIOD(DAY/WEEK/MONTH)] [t/TAG]...'
```
Comment (louislai) [inline docs/UserGuide.md]: Probably will have to split out this into a few commands, otherwise there should be an option for the user to update start date & end date, deadline or name, or tags independently. E.g user may just want to update deadline of task without changing name

**Comment by louislai** [comment]
@Melvin-Tan

File: docs/UserGuide.md [hunk 6]
```diff
@@ -41,138 +31,225 @@ By : `Team SE-EDU`  &nbsp;&nbsp;&nbsp;&nbsp; Since: `Jun 2016`  &nbsp;&nbsp;&nbs
...
+
+Changes the location for the storage file used in this system
+Format: 'setstoragelocation NEWFILELOCATION'
+
+> * All data will be moved to the new file location.
+> * If the file does not exist, the file will be created.
+> * The old file will be removed.
+
+Example:
+* 'setstoragelocation data/savefile.txt'
```
Comment (louislai) [inline docs/UserGuide.md]: a bit too wordy as a command, either `store` or `save` would work as well

File: docs/UserGuide.md [hunk 7]
```diff
@@ -41,138 +31,225 @@ By : `Team SE-EDU`  &nbsp;&nbsp;&nbsp;&nbsp; Since: `Jun 2016`  &nbsp;&nbsp;&nbs
 > * Items with `...` after them can have multiple instances.
 > * Parameters can be in any order.
 
+### 2.0. Viewing tasks
+
+> * A list of tasks will always be displayed.
```
Comment (louislai) [inline docs/UserGuide.md:36]: I was thinking we can combine this with find commands through a uniform syntax, e.g `list`. So `list` will display all tasks, while `list <term>` acts like a find

File: docs/UserGuide.md [hunk 8]
```diff
@@ -41,138 +31,225 @@ By : `Team SE-EDU`  &nbsp;&nbsp;&nbsp;&nbsp; Since: `Jun 2016`  &nbsp;&nbsp;&nbs
...
 
-### 2.5. Finding all persons containing any keyword in their name: `find`
+### 2.4. Finding all tasks containing any keyword in their name or tag or date: `find`
 
-Finds persons whose names contain any of the given keywords.<br>
-Format: `find KEYWORD [MORE_KEYWORDS]`
+Finds tasks whose names contain any of the given keywords.<br>
+Format: `find [KEYWORDS] [t/] [d/] [n/]`
 
 > * The search is case sensitive. e.g `hans` will not match `Hans`
```
Comment (Melvin-Tan) [inline docs/UserGuide.md]: Not ok, try to make it case insensitive.

File: docs/UserGuide.md [hunk 9]
```diff
@@ -41,138 +31,225 @@ By : `Team SE-EDU`  &nbsp;&nbsp;&nbsp;&nbsp; Since: `Jun 2016`  &nbsp;&nbsp;&nbs
...
-> * Only the name is searched.
+> * By default only the name is searched.
+> * Adding 't/' will search by tag instead.
+> * Adding 'd/' will search by end date instead.
+> * Adding 'n/' will still search by name.
+> * If no keyword is entered, the list of all tasks is displayed.
+> * 't/', 'd/' and 'n/' can be combined to search by multiple fields
 > * Only full words will be matched e.g. `Han` will not match `Hans`
-> * Persons matching at least one keyword will be returned (i.e. `OR` search).
+> * tasks matching at least one keyword will be returned (i.e. `OR` search).
```
Comment (Melvin-Tan) [inline docs/UserGuide.md]: "tasks" change to "Tasks"

File: docs/UserGuide.md [hunk 10]
```diff
@@ -41,138 +31,225 @@ By : `Team SE-EDU`  &nbsp;&nbsp;&nbsp;&nbsp; Since: `Jun 2016`  &nbsp;&nbsp;&nbs
...
     e.g. `Hans` will match `Hans Bo`
 
 Examples:
 
-* `find John`<br>
-  Returns `John Doe` but not `john`
-* `find Betsy Tim John`<br>
-  Returns Any person having names `Betsy`, `Tim`, or `John`
+* `find Assignment`<br>
+  Returns `Assignment 1` but not `assignment 2`
```
Comment (Melvin-Tan) [inline docs/UserGuide.md]: Based on the case insensitive search that i mentioned above. Need to update

File: docs/UserGuide.md [hunk 11]
```diff
@@ -41,138 +31,225 @@ By : `Team SE-EDU`  &nbsp;&nbsp;&nbsp;&nbsp; Since: `Jun 2016`  &nbsp;&nbsp;&nbs
...
-  `select 2`<br>
-  Selects the 2nd person in the address book.
-* `find Betsy` <br>
-  `select 1`<br>
-  Selects the 1st person in the results of the `find` command.
+* `add Test`<br>
+  `undo`<br>
+  Undoes adding Test to the todo list.
+* `add Assignment` <br>
+  `add Project`
```
Comment (Melvin-Tan) [inline docs/UserGuide.md]: Forgot to add a line break here

**Comment by louislai** [comment]
Btw do refer to the relevant issue in future https://github.com/CS2103JAN2017-W13-B3/main/issues/1

File: docs/UserGuide.md [hunk 12]
```diff
@@ -41,138 +31,225 @@ By : `Team SE-EDU`  &nbsp;&nbsp;&nbsp;&nbsp; Since: `Jun 2016`  &nbsp;&nbsp;&nbs
...
+  `undo 2`<br>
+  Undoes both commands.
```
Comment (Melvin-Tan) [inline docs/UserGuide.md]: Not sure if "Undo" or "Undoes" here.. i think "Undo" is more correct. (Same for the one at line 148)

**Review by louislai — CHANGES_REQUESTED** [review]

File: docs/UserGuide.md [hunk 13]
```diff
@@ -41,138 +31,225 @@ By : `Team SE-EDU`  &nbsp;&nbsp;&nbsp;&nbsp; Since: `Jun 2016`  &nbsp;&nbsp;&nbs
...
 
-### 2.8. Clearing all entries : `clear`
+### 2.9. Redo a command : `redo`
 
-Clears all entries from the address book.<br>
-Format: `clear`
+Redoes previously undone commands by the user.<br>
```
Comment (Melvin-Tan) [inline docs/UserGuide.md]: Not sure if "Redo" or "Redoes" here too. I think it should be "Redo"

File: docs/UserGuide.md [hunk 14]
```diff
@@ -41,138 +31,225 @@ By : `Team SE-EDU`  &nbsp;&nbsp;&nbsp;&nbsp; Since: `Jun 2016`  &nbsp;&nbsp;&nbs
...
+> The number **must be a positive integer** 1, 2, 3, ...
+> The number must be less than or equal to the number of commands undone.
 
-Exits the program.<br>
-Format: `exit`
+Examples:
+
+* `add Test`<br>
+  `undo`<br>
+  `redo`
```
Comment (Melvin-Tan) [inline docs/UserGuide.md]: You missed a line break here

File: docs/UserGuide.md (hunk 5 above)
Comment (Rinder5) [reply docs/UserGuide.md]: Anything with [] is optional, should be stated in the beginning...

File: docs/UserGuide.md [hunk 15]
```diff
@@ -41,138 +31,225 @@ By : `Team SE-EDU`  &nbsp;&nbsp;&nbsp;&nbsp; Since: `Jun 2016`  &nbsp;&nbsp;&nbs
...
+* `add Assignment` <br>
+  `add Project`<br>
+  `undo 2`<br>
+  `redo`<br>
+  Redoes `add Assignment`.
 
 ### 2.10. Saving the data
 
-Address book data are saved in the hard disk automatically after any command that changes the data.<br>
+todo list data are saved in the hard disk automatically after any command that changes the data.<br>
```
Comment (Melvin-Tan) [inline docs/UserGuide.md]: "todo" change to "Todo"

File: docs/UserGuide.md (hunk 6 above)
Comment (Rinder5) [reply docs/UserGuide.md]: possible, but the user should really only need to do this once, so it's fine if its wordy. Also helps against accidentally doing it in the future

File: docs/UserGuide.md [hunk 16]
```diff
@@ -41,138 +31,225 @@ By : `Team SE-EDU`  &nbsp;&nbsp;&nbsp;&nbsp; Since: `Jun 2016`  &nbsp;&nbsp;&nbs
...
 There is no need to save manually.
 
-## 3. FAQ
+### 2.11. Viewing previous commands and accessing them : 'history'
 
-**Q**: How do I transfer my data to another Computer?<br>
-**A**: Install the app in the other computer and overwrite the empty data file it creates with
-       the file that contains the data of your previous Address Book folder.
+Shows previous commands entered
```
Comment (Melvin-Tan) [inline docs/UserGuide.md]: Missed a line break here (and a fullstop)

File: docs/UserGuide.md [hunk 17]
```diff
@@ -41,138 +31,225 @@ By : `Team SE-EDU`  &nbsp;&nbsp;&nbsp;&nbsp; Since: `Jun 2016`  &nbsp;&nbsp;&nbs
...
+Format: 'history'
 
-## 4. Command Summary
+> * previous commands are listed in order from latest command to earlier command
```
Comment (Melvin-Tan) [inline docs/UserGuide.md]: "previous" -> "Previous"

File: docs/UserGuide.md (hunk 4 above)
Comment (Rinder5) [reply docs/UserGuide.md]: Similar to previous comment, anything with [] is optional

File: docs/UserGuide.md [hunk 18]
```diff
@@ -41,138 +31,225 @@ By : `Team SE-EDU`  &nbsp;&nbsp;&nbsp;&nbsp; Since: `Jun 2016`  &nbsp;&nbsp;&nbs
...
+> * alternatively, pressing on the 'up' and 'down' arrow keys on the keyboard will cycle through the commands previously entered
```
Comment (Melvin-Tan) [inline docs/UserGuide.md]: "alternatively" -> "Alternatively"

File: docs/UserGuide.md [hunk 19]
```diff
@@ -41,138 +31,225 @@ By : `Team SE-EDU`  &nbsp;&nbsp;&nbsp;&nbsp; Since: `Jun 2016`  &nbsp;&nbsp;&nbs
...
+
+Examples:
+* 'add Test' <br>
+  'history' <br>
+  Shows 'add Test' in the list.
+* 'add Test' <br>
+  you press on the 'up' arrow key. <br>
```
Comment (Melvin-Tan) [inline docs/UserGuide.md]: "you" -> "You"
Might want to wrap the 'up' arrow key in something like this <kbd>Enter</kbd>

File: docs/UserGuide.md (hunk 3 above)
Comment (Rinder5) [reply docs/UserGuide.md]: Not sure what you mean here by mismatched example.

Also, parameters can be entered in any order, so it doesn't matter which one is listed first.

File: docs/UserGuide.md [hunk 20]
```diff
@@ -41,138 +31,225 @@ By : `Team SE-EDU`  &nbsp;&nbsp;&nbsp;&nbsp; Since: `Jun 2016`  &nbsp;&nbsp;&nbs
...
+  Shows 'add Test' in your input text field.
 
-* **Add**  `add NAME p/PHONE_NUMBER e/EMAIL a/ADDRESS [t/TAG]...` <br>
-  e.g. `add James Ho p/22224444 e/jamesho@gmail.com a/123, Clementi Rd, 1234665 t/friend t/colleague`
+### 2.12. Add alias for commands: 'add alias'
+
+Adds an alias for a command
```
Comment (Melvin-Tan) [inline docs/UserGuide.md]: Missed a line break here (and a fullstop)

File: docs/UserGuide.md [hunk 21]
```diff
@@ -41,138 +31,225 @@ By : `Team SE-EDU`  &nbsp;&nbsp;&nbsp;&nbsp; Since: `Jun 2016`  &nbsp;&nbsp;&nbs
...
+* 'add Test' <br>
+  'history' <br>
+  Shows 'add Test' in the list.
+* 'add Test' <br>
+  you press on the 'up' arrow key. <br>
+  Shows 'add Test' in your input text field.
 
-* **Add**  `add NAME p/PHONE_NUMBER e/EMAIL a/ADDRESS [t/TAG]...` <br>
-  e.g. `add James Ho p/22224444 e/jamesho@gmail.com a/123, Clementi Rd, 1234665 t/friend t/colleague`
+### 2.12. Add alias for commands: 'add alias'
```
Comment (Melvin-Tan) [inline docs/UserGuide.md]: "'add alias'" -> "'addalias'"? Not sure if you intended for this o.o?

File: docs/UserGuide.md [hunk 22]
```diff
@@ -41,138 +31,225 @@ By : `Team SE-EDU`  &nbsp;&nbsp;&nbsp;&nbsp; Since: `Jun 2016`  &nbsp;&nbsp;&nbs
...
+> * Inputting a value for command will only show all aliases for that command.
+> * Otherwise will show all aliases.
+> * Lists aliases in the format 'COMMAND:ALIAS'.
+
+Example:
+* 'addalias history hs' <br>
+  'viewaliases' <br>
+  Shows 'history:hs' in the list.
+
+### 2.15. Change storage location for data: 'change storage location'
```
Comment (Melvin-Tan) [inline docs/UserGuide.md]: Should 'change storage location' be 'setstoragelocation'?

File: docs/UserGuide.md (hunk 6 above)
Comment (louislai) [reply docs/UserGuide.md]: it may happen multiple times throughout the app usage, not just once I think. These commands are like variable names or method names, basically you would want to make them as readable as possible. So `setstoragelocation` in method name format would be `setStorageLocation` or `set_storage_location. Just setting to a single verb would be easier, and more consistent as well

File: docs/UserGuide.md (hunk 5 above)
Comment (louislai) [reply docs/UserGuide.md]: I see. In the future we might wanna get rid of <key>/<value> syntax to come up with a more natural format tho. `add <task> d/<date>` can be `add <task> by <date>`

**Review by Melvin-Tan — CHANGES_REQUESTED** [review]

File: docs/UserGuide.md (hunk 7 above)
Comment (Melvin-Tan) [reply docs/UserGuide.md:36]: Let's raise this again in the next meeting..? This is a design issue that I don't wanna debate over in a PR review.

File: docs/UserGuide.md (hunk 3 above)
Comment (louislai) [reply docs/UserGuide.md]: Oh i was commenting on the description I guess. The comment is **`add`**` Try Out Todo List  d/11-12-2011` , but the description did not meniton anyhting abt the date. 

Even if the parameter can be entered in any order, it's usually preferable to put the example in the natural order that the user would input, which will be start date followed by end date. That is what I believe

File: docs/UserGuide.md (hunk 6 above)
Comment (Melvin-Tan) [reply docs/UserGuide.md]: I think both of you have valid points. In this case i think since the argument contains a filename, we can make the command word short since it is understood. So probably follow Louis idea might be better.

File: docs/UserGuide.md (hunk 3 above)
Comment (Melvin-Tan) [reply docs/UserGuide.md]: This brings us back to our idea of making the commands more fluid, so things like d/ n/ sd/ will all go away and be replaced by words like "from"/"by" etc. 
For now, this is okay for v0.0.

File: docs/UserGuide.md (hunk 4 above)
Comment (Melvin-Tan) [reply docs/UserGuide.md]: So long as you flag this in your example, it should be ok.

File: docs/UserGuide.md (hunk 1 above)
Comment (Melvin-Tan) [reply docs/UserGuide.md]: To meeting. Same as the comment on line 217.

File: docs/UserGuide.md (hunk 4 above)
Comment (Rinder5) [reply docs/UserGuide.md]: you mean the [] or having an example for undo 5?

the [] is listed at the beginning.
undo has an example for single use and multiple use versions.

Comment (Melvin-Tan) [reply docs/UserGuide.md]: Yup "undo has an example of single use" is what I am looking for!

File: docs/UserGuide.md [hunk 23]
```diff
@@ -1,35 +1,25 @@
-# AddressBook Level 4 - User Guide
+# TuLuist - User Guide
```
Comment (Melvin-Tan) [inline docs/UserGuide.md]: I think it's ToLuist instead of TuLuist? Just flagging this so you and @louislai notice...

File: docs/UserGuide.md (hunk 3 above)
Comment (louislai) [reply docs/UserGuide.md]: fair point. Agree

File: docs/UserGuide.md (hunk 7 above)
Comment (louislai) [reply docs/UserGuide.md:36]: Issue created here https://github.com/CS2103JAN2017-W13-B3/main/issues/16

File: docs/UserGuide.md (hunk 1 above)
Comment (louislai) [reply docs/UserGuide.md]: Issue created here https://github.com/CS2103JAN2017-W13-B3/main/issues/17

File: docs/UserGuide.md (hunk 8 above)
Comment (louislai) [reply docs/UserGuide.md]: ^

File: docs/UserGuide.md [hunk 24]
```diff
@@ -41,146 +31,253 @@ By : `Team SE-EDU`  &nbsp;&nbsp;&nbsp;&nbsp; Since: `Jun 2016`  &nbsp;&nbsp;&nbs
...
+  `history` <br>
+  Shows `add Test` in the list.
+* `add Test` <br>
+  you press on the <kbd>up</kbd>kbd> arrow key. <br>
+  Shows `add Test` in your input text field.
 
-* **Add**  `add NAME p/PHONE_NUMBER e/EMAIL a/ADDRESS [t/TAG]...` <br>
-  e.g. `add James Ho p/22224444 e/jamesho@gmail.com a/123, Clementi Rd, 1234665 t/friend t/colleague`
+### 2.12. Add alias for commands: `add alias`
+
```
Comment (louislai) [inline docs/UserGuide.md]: @Rinder5 refer to the discussion over here https://github.com/CS2103JAN2017-W13-B3/main/issues/17

**Comment by louislai** [comment]
need to merge this soon. pls handle all the comments

**Comment by louislai** [comment]
@Rinder5

**Comment by Rinder5** [comment]
K, changed a bunch of stuff

File: docs/UserGuide.md (hunk 1 above)
Comment (Rinder5) [reply docs/UserGuide.md]: I changed how alias works. See if you like it.

File: docs/UserGuide.md (hunk 2 above)
Comment (Rinder5) [reply docs/UserGuide.md]: This one i didn't end up changing. We can discuss tomorrow which one is better.

**Comment by Rinder5** [comment]
resolve #17 
resolve #1

File: docs/UserGuide.md (hunk 7 above)
Comment (louislai) [reply docs/UserGuide.md:36]: @Rinder5 sorry for keeping asking for more stuff :S can put a `list` here as well. Just list all tasks (referred to in #16)

File: docs/UserGuide.md [hunk 25]
```diff
@@ -41,146 +31,257 @@ By : `Team SE-EDU`  &nbsp;&nbsp;&nbsp;&nbsp; Since: `Jun 2016`  &nbsp;&nbsp;&nbs
...
+Format: `alias ALIAS PHRASE`
+
+> * Once added, alias can be used instead of the phrase to perform operations.
+> * The phrase can be multiple words long
+
+Example:
+* `alias hs history` <br>
+  `hs` <br>
+  Shows `alias hs history` in the list.
+* `alias addTaskNamedTest add Test` <br>
```
Comment (louislai) [inline docs/UserGuide.md:209]: hmm kinda different from what we talked abt in #17 . should have quotes for the multi-word aliased command (so `alias addTaskNamedTest "add Test"`).

**Comment by louislai** [comment]
@Rinder5 got some minor issues left. I'll just help you edit and merge in after

File: docs/UserGuide.md (hunk 25 above)
Comment (Rinder5) [reply docs/UserGuide.md:209]: I don't really see the point of the quotes. Theres nothing left after the alias name, so we can just use anything there to be the command

File: docs/UserGuide.md (hunk 7 above)
Comment (Rinder5) [reply docs/UserGuide.md:36]: its covered in find already. You can just type `find` without any parameters to show everything

File: docs/UserGuide.md (hunk 25 above)
Comment (louislai) [reply docs/UserGuide.md:209]: Actually I do agree with u. The quote just makes the unaliased component more obvious (to the user). @Melvin-Tan thoughts?

File: docs/UserGuide.md (hunk 7 above)
Comment (louislai) [reply docs/UserGuide.md:36]: If the user just wants to see everything, `list` is sth they would think of first before `find` though
